app/src/main/java/ca/lakeeffect/scoutingapp/MainActivity.java

```diff
@@ -447,7 +450,13 @@ public void run() {
         System.out.println(labels.toString());
         System.out.println(data.toString());
         String[] out = {data.toString(), labels.toString()};
-        return out;
+        byte[] data2 = out[0].getBytes(Charset.forName("UTF-8"));
+        byte[] labels2 = out[1].getBytes(Charset.forName("UTF-8"));
```

Comment:
I think this would work better if you worked with the data and labels variable directly, removing the out variable entirely.

Also, I suggest renaming `data2` and `labels2` to something like `dataBytes`